engine/src/main/java/com/datatorrent/stram/StreamingAppMasterService.java

```diff
@@ -870,6 +873,21 @@ private void execute() throws YarnException, IOException
       }
 
       numRequestedContainers += containerRequests.size() - removedContainerRequests.size();
+
```

Comment:
1. What do you mean overloaded? If the name is misleading we can rename it (it is private) to indicate what it is supposed to be doing including the new code in this PR suggested by me.

2. The end result is the same. When onContainerStopped is called the contaierIds are removed from pendingContainersToKill and if it is not called (the case you are saying) we will detect the containers that have been in pendingContainersToKill for too long (more than NODE_MANAGER_KILL_CONTAINER_TIMEOUT).

I suggest you reconsider